src/main/webapp/app/text-assessment/text-assessment.component.ts

```diff
@@ -0,0 +1,132 @@
+        const submissionId = Number(this.route.snapshot.paramMap.get('submissionId'));
+
+        this.assessmentsService.getFeedbackDataForExerciseSubmission(exerciseId, submissionId).subscribe(participation => {
+            this.participation = participation;
+            this.submission = <TextSubmission>this.participation.submissions[0];
+            this.exercise = <TextExercise>this.participation.exercise;
+            this.result = this.participation.results[0];
+            this.assessments = this.result.feedbacks;
+            if (!this.assessments) {
+                this.assessments = [];
+            }
+            this.busy = false;
+        });
+    }
+
+    public addAssessment(assessmentText: string): void {
+        const assessment = new Feedback();
+        assessment.reference = assessmentText;
+        assessment.credits = 0;
+        this.assessments.push(assessment);
+        this.checkScoreBoundaries();
+    }
+
+    public deleteAssessment(assessmentToDelete: Feedback): void {
+        this.assessments = this.assessments.filter(elem => elem !== assessmentToDelete);
+        this.checkScoreBoundaries();
+    }
+
+    public save(): void {
+        this.checkScoreBoundaries();
+        if (!this.assessmentsAreValid) {
+            this.jhiAlertService.error('arTeMiSApp.textAssessment.invalidAssessments');
+            return;
+        }
+
+        this.assessmentsService.save(this.assessments, this.exercise.id, this.result.id).subscribe(response => {
+            this.result = response.body;
+            this.jhiAlertService.success('arTeMiSApp.textAssessment.saveSuccessful');
+        });
+    }
+
+    public submit(): void {
+        this.checkScoreBoundaries();
+        if (!this.assessmentsAreValid) {
+            this.jhiAlertService.error('arTeMiSApp.textAssessment.invalidAssessments');
+            return;
+        }
+
+        this.assessmentsService.submit(this.assessments, this.exercise.id, this.result.id).subscribe(response => {
+            response.body.participation.results = [response.body];
+            this.result = response.body;
+            this.jhiAlertService.success('arTeMiSApp.textAssessment.submitSuccessful');
+        });
+    }
+
+    public previous(): void {
+        this.router.navigate(['text', this.exercise.id, 'assessment']);
+    }
+
+    /**
+     * Calculates the total score of the current assessment.
+     * Returns an error if the total score cannot be calculated
+     * because a score is not a number/empty.
+     */
+    public checkScoreBoundaries() {
+        if (!this.assessments || this.assessments.length === 0) {
+            this.totalScore = 0;
+            this.assessmentsAreValid = true;
+            return;
+        }
+
+        const credits = this.assessments.map(assessment => assessment.credits);
+
+        if (!credits.every(credit => credit !== null)) {
+            this.invalidError = 'The score field must be a number and can not be empty!';
```

Comment:
You are not checking that credit is a number. I think the condition should be something like
```javascript
if (!credits.every(credit => credit !== null && !isNaN(credit))) {
```

Reply:
👍